tests/components/template/test_sensor.py

```diff
@@ -405,6 +452,32 @@ def test_setup_valid_device_class(self):
     assert ("Error rendering attribute test_attribute") in caplog.text
 
 
+async def test_invalid_availability_template_keeps_component_available(hass, caplog):
+    """Test that an invalid availability keeps the device available."""
+
+    await async_setup_component(
+        hass,
+        "sensor",
+        {
+            "sensor": {
+                "platform": "template",
+                "sensors": {
+                    "my_sensor": {
+                        "value_template": "{{ states.sensor.test_state.state }}",
+                        "availability_template": "{{ x - 12 }}",
+                    }
+                },
+            }
+        },
+    )
+
+    await hass.async_start()
+    await hass.async_block_till_done()
+
+    assert hass.states.get("sensor.my_sensor") != STATE_UNAVAILABLE
```

Comment:
You need to verify the state. Not the state object against unavailable. Better to verify that it is the value that you expect.

Reply:
OK. But this uses the standard `def available()` mechanism that all components use. i.e. if `available()` returns `(bool) True`, the device operated as normal, with it's normal `state`. If  `available()` returns `(bool) False` the device is `STATE_UNAVAILABLE`.

What other way is there to test the result of the `available()` method through the `hass.states...` mechanism?

Reply:
Done